src/H5Eint.c

```diff
@@ -1791,3 +1790,79 @@ H5E_dump_api_stack(void)
 
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5E_dump_api_stack() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5E_pause_stack
+ *
+ * Purpose:     Pause pushing errors on the default error stack.
+ *
+ *              Generally used via the H5E_PAUSE_ERRORS / H5E_RESUME_ERRORS
+ *              macros.
+ *
+ *              When one needs to temporarily disable recording errors while
+ *              trying something that's likely or expected to fail.  The code
+ *              to try can be nested between these macros like:
+ *
+ *              H5E_PAUSE_ERRORS {
+ *                  ...stuff here that's likely to fail...
+ *              } H5E_RESUME_ERRORS
+ *
+ *              Warning: don't break, return, or longjmp() from the block of
+ *              code or the error reporting won't be properly restored!
+ *
+ * Return:      none
+ *
+ *-------------------------------------------------------------------------
+ */
+void
+H5E_pause_stack(void)
+{
+    H5E_stack_t *estack = H5E__get_my_stack();
+
+    FUNC_ENTER_NOAPI_NOERR
+
+    assert(estack);
+
+    /* Increment pause counter */
+    estack->paused++;
+
+    FUNC_LEAVE_NOAPI_VOID
+} /* end H5E_pause_stack() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5E_resume_stack
+ *
+ * Purpose:     Resume pushing errors on the default error stack.
+ *
+ *              Generally used via the H5E_PAUSE_ERRORS / H5E_RESUME_ERRORS
+ *              macros.
+ *
+ *              When one needs to temporarily disable recording errors while
+ *              trying something that's likely or expected to fail.  The code
+ *              to try can be nested between these macros like:
+ *
+ *              H5E_PAUSE_ERRORS {
+ *                  ...stuff here that's likely to fail...
+ *              } H5E_RESUME_ERRORS
+ *
+ *              Warning: don't break, return, or longjmp() from the block of
+ *              code or the error reporting won't be properly restored!
+ *
+ * Return:      none
+ *
+ *-------------------------------------------------------------------------
+ */
+void
+H5E_resume_stack(void)
+{
+    H5E_stack_t *estack = H5E__get_my_stack();
+
+    FUNC_ENTER_NOAPI_NOERR
+
+    assert(estack);
+
+    /* Decrement pause counter */
+    estack->paused--;
```

Comment:
I'm thinking it would probably be a good idea to make sure this isn't 0 before decrementing it so that a `resume_stack` call doesn't accidentally pause the stack.

Reply:
Sure, I'll add an assert.